aspnet/mvc/models/validation.rst

```diff
@@ -1,8 +1,230 @@
-.. include:: /../common/stub-topic.txt
+Model Validation 
+================
+By `Rachel Appel <http://github.com/rachelappel>`_
 
-|stub-icon| Model Validation
-============================
+In this article:
 
-.. include:: /../common/stub-notice.txt
+.. contents:: Sections
+  :local:
+  :depth: 1
 
-.. _issue: https://github.com/aspnet/Docs/issues/116
+Introduction to model validation
+--------------------------------
+Before an app stores data in a database, the app must validate the data. Data must be scrubbed for security, verified that it is appropriately formatted by type and size, and it must conform to your rules. Validation is necessary although it can be redundant and tedious to implement. In MVC, validation happens on both the client and server. 
+
+Fortunately, .NET has abstracted validation into data annotations. These attributes contain validation code, thereby reducing the amount of code you must write. 
+
+Data annotations
+----------------
+Data annotations are a way to configure model validation so it's similar conceptually to validation on fields in tables. This includes constraints such as assigning data types or required fields. Other types of validation include applying patterns to data to enforce business rules, such as a credit card, phone number, or email address. Data annotations make enforcing these requirements much simpler and easier to use.
+
+Below is an annotated ``Movie`` model from an app that stores information about movies and TV shows. Most of the properties are required, the ``Id`` property is a primary key, and several string properties have length requirements. Additionally, there is a numeric range in place for the ``Price`` property from 0 to $999.99, along with a custom annotation.
+
+.. literalinclude:: validation/sample/Movie.cs
+   :language: c#
+   :lines: 7-34
+   :dedent: 4
+
+Simply reading through the model reveals the rules about data for this app, making it easier to maintain the code. Below are several popular built-in data annotations:
+
+- ``[CreditCard]``: Specifies the property as a credit card type.
+- ``[Compare]``: Compares two properties in a model. It works with various types, such as int, string, or date. 
+- ``[EmailAddress]``: Specifies the property with an email format.
+- ``[Phone]``: Specifies the property as a telephone format.
+- ``[Range]``: Sets the minimum and maximum values for multiple types of properties.
+- ``[RegularExpression]``: Forces the data to match a regular expression.
+- ``[Required]``: Makes a property required.
+- ``[StringLength]``: The maximum length for a string field.
+- ``[Url]``: Specifies the property as a URL.
```

Comment:
Reword all use of "Specifies ... as a ..." e.g. "Validates the property has the format of a URL."